unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -1073,16 +1075,39 @@ protected bool IsCollided()
 			return GameObject.FindObjectsOfType<SimObj>();
         }
 
+        public void ResetObjectFilter() {
+            this.simObjFilter = null;
+            actionFinished(true);
+        }
+        public void SetObjectFilter(string[] objectIds) {
+            SimObjPhysics[] simObjects = GameObject.FindObjectsOfType<SimObjPhysics>();
+            simObjFilter = new HashSet<SimObjPhysics>();
+            foreach(var simObj in simObjects) {
+                if(Array.IndexOf(objectIds, simObj.ObjectID) >= 0) {
```

Comment:
having a dictionary of the simobjects by id could be good so you don't have to look in the `objectIds` array every time, the array is probably often small so no big deal.

Reply:
A hashset seems like the right structure here and this array lookup is done so rarely (only on setting the filter
on each request to retrieve the actual object.  Doing this Array.IndexOf() is only done when you set the filter